### PR TITLE
Single Column Model "Replay" Option

### PR DESCRIPTION
**Pull request by bogensch**
Modifications to allow for the Single Column Model (SCM) "Replay" option.  This will allow the user to run the full E3SM model and extract forcing for a desired location that can then be used to drive the E3SM SCM (with spectral element dynamical core only).  Note that code modifications provided in this PR will only reproduce the behavior of a column.  This PR will not yet produce bit-4-bit reproduction of a column with the SCM.  A future PR will address this.  

This branch passes all e3sm_developer tests

### CODE REVIEW

File: components/cam/bld/configure [hunk 1]
```diff
@@ -1223,10 +1223,10 @@ if (defined $opts{'camiop'}) {
 }
 my $camiop = $cfg_ref->get('camiop') ? "ON" : "OFF";
 
-# The only dycore supported in CAMIOP mode is Eulerian
-if ($camiop eq 'ON' and $dyn_pkg ne 'eul') {
+# The only dycore supported in CAMIOP mode is Spectral Element
```
Comment (AaronDonahue) [inline components/cam/bld/configure]: Are we removing support for the Eulerian dycore altogether?  Or is it that the replay option will only work with SE and Eulerian will still work as usual?

Comment (bogensch) [reply components/cam/bld/configure]: Just the replay option will work with SE core.  Eulerian core is still the default option for the E3SM SCM for the time being.

Comment (PeterCaldwell) [reply components/cam/bld/configure]: Note that I'm hoping that we can eventually ditch/delete the Eulerian dycore completely, but we need the SE dycore to be made the default first.

**Comment by wlin7** [comment]
Merged to next. Conflict in scamMod.F90 resolved during merge from command line.

**Comment by wlin7** [comment]
Merged.

**Comment by rljacob** [comment]
This should not have been merged.  It caused diffs in SMS_R_Ld5.T42_T42.FSCM5A97.   @wlin7 why did you think you were clear to merge to master?

**Comment by wlin7** [comment]
Sorry @rljacob . I have an email problem recently and some emails screwed up when arriving at my mailbox. I think I saw the integrator status as 'Open' for both next and master yesterday and I didn't expect this PR to be non-BFB. I should have double checked this test specifically.

**Comment by wlin7** [comment]
Hi @bogensch , can you please review what could be causing non-BFB? To me, the changes that could impact the simulation results are all with CAM_IOP, which is not on in the integration test.

**Comment by wlin7** [comment]
@rljacob , should I revert this merge, or leave it in next and master while working to fix it or determine if the DIFF can be justified? There will be no other SCM-related commit until this one is resolved.

**Comment by rljacob** [comment]
revert it from master for now.

**Comment by bogensch** [comment]
Yes, @wlin7 will be looking into it today.

**Comment by wlin7** [comment]
@bogensch , looks like the differences could be due to the addition/subtraction of some fields to history file, which does not affect the simulation itself. Is that so?

**Comment by bogensch** [comment]
@wlin7 on my end the only variable I see different in my tests is OMEGA.  This is because of the change in cam_diagnostics.F90 where the definition of OMEGA is changed to be consistent with that of the state%omega.  Thus this diff is due to a diagnostic change, not a change in the simulation itself.  Sorry, I missed that when doing the analysis of my tests.

**Comment by wlin7** [comment]
Thanks for looking into it, @bogensch . I still find it hard to explain why the change in cam_diagnostics.F90 could cause differences to omega in output. In EUL-SCM mode, wfld and omega are currently the same: wfld was assigned to omga in eul/stepon.F90 in SCM mode.

The test failure is indeed due to DIFF in cam.h0. I feel like changes in history_defaults.F90 may be response, because some variables are dropped and some others added by default.

The good thing is you also agree the simulation results are not affected.  @rljacob agreed (during IG telecon) that we don't need to revert the merge. We may just request to bless the test if the DIFF can all be justified. I will also do some tests before submitting the request.

**Comment by wlin7** [comment]
Hi @bogensch , you are right. I verified the simulations before and after the merge. The simulations are BFB. The differences are only in the omega field. The changes in history_defaults.F90 do not matter here. It would only affect h1 file.

Can you explain the differences in omega? They start to differ from the 3rd step. I initially thought the differences would be just the placement of the vertical position: at half layer vs interface level. But it does not appear so by looking at the values. 

With wfld output as omega, the field would strictly be equal to the prescribed value, which would be expected by SCM modelers. With the current modification, that changes. How to justify such a change? Although it is just a diagnostic output, it could confusion because the omega in SCM output would now differ (very noticeably) from that in the forcing file, and the differences are not simply due to interpolation. 
 
We can still leave the merge there. If you think we should make the modification, we can just bless the DIFF. Otherwise, we can commit a new PR to reverse just that particular change. Thanks.

**Comment by bogensch** [comment]
@wlin7 .  Yes, I just looked at the output and drew the same conclusions.  

When I was running/testing the REPLAY option in the SE SCM, I noticed I was not getting b4b results at the first time step because the prescribed omega was not successfully ever passed to the state%omega (only CLUBB uses this value).  I fixed that issue in SE SCM and noticed that it didn't matter which value was output (wfld vs. state%omega) and removed the "if (single_column)" switch for this diagnostic value as I saw it as redundant.  I had assumed that the Eulerian code WAS successfully passing the prescribed omega value to state, but it appears that it does not.  

Thus, this appears to be a legacy bug related to the Eulerian SCM.  I would vote to keep the code as it is and I will look into addressing this bug in my next PR.

**Comment by bogensch** [comment]
Sounds good @wlin7 .  I am hoping to have the fix within a week or two.

**Pull request merged**